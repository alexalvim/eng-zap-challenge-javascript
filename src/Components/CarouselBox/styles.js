import styled from 'styled-components';

import Spaces from '../../spaces';
import Typo from '../../typo';

export const ContentWrapper = styled.div`
  border-radius: 5px;
  box-shadow: 0px 3px 5px rgba(0,0,0,0.2);
  display: flex;
  flex-direction: column;
  margin: ${Spaces.double} 0;
  overflow: hidden;

  @media(min-width: ${Spaces.tabletBreakpoint}){
    flex-direction: row;
  }
`;

export const CarouselContainer = styled.div`
  width: 100%;

  .slick-arrow:before {
    opacity: 1;
    text-shadow: 1px 3px 3px rgba(0,0,0,0.5);
  }

  .slick-arrow.slick-disabled:before {
    opacity: .5;
  }

  .slick-next {
    right: 5px;
    z-index: 1;
  }

  .slick-prev {
    left: 5px;
    z-index: 1;
  }

  @media(min-width: ${Spaces.tabletBreakpoint}){
    width: 350px;
  }
`;

export const ImageContainer = styled.div`
  align-items: center;
  display: flex !important;
  height: 300px;
  justify-content: center;
  overflow: hidden;
  position: relative;

  > img {
    width: auto;
    height: 300px;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
  }

  @media(min-width: ${Spaces.tabletBreakpoint}){
    max-width: 350px;
  }
`;

export const TextHolder = styled.div`
  display: flex;
  flex-direction: column;
  padding: ${Spaces.base};
`;

export const titleWrapperStyles = `
  font-size: ${Typo.large};
  margin-bottom: ${Spaces.base};
`;