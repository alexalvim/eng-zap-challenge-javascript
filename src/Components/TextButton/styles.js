import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';
import Typo from '../../typo';

export const Button = styled.button`
  background: none;
  border: none;
  cursor: pointer;
  font-size: ${Typo.medium}
  font-weight: 500;
  padding: 0;

  &:hover {
    text-decoration: underline;
  }

  ${({activePortal}) => activePortal && `color: ${Colors.portalMainColor[activePortal]};`}
`;