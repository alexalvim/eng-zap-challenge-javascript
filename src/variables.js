export const PROPERTIES_PER_PAGE = 20;

export const PORTALS = {
  VIVA_REAL: 'VivaReal',
  ZAP: 'Zap',
}

export const BOUDING_BOX_NUMBERS = {
  MIN_LAT: -23.568704,
  MIN_LON: -46.693419,
  MAX_LAT: -23.546686,
  MAX_LON: -46.641146
}