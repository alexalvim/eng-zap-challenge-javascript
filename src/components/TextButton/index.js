import React from 'react';

import { Button } from './styles'

export default ({ label, onClick, activePortal, wrapperStyles }) => (
  <Button
    styles={wrapperStyles}
    activePortal={activePortal}
    onClick={onClick}>
    {label}
  </Button>
)