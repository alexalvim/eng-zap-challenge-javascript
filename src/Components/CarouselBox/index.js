import React from 'react';
import Slider from 'react-slick';

import TextButton from '../TextButton';
import {
  ContentWrapper,
  CarouselContainer,
  ImageContainer,
  TextHolder,
  titleWrapperStyles
} from './styles'

export default ({ images, activePortal, title, children }) => {
  const settings = {
    infinite: false,
    speed: 500,
    slidesToShow: 1,
    slidesToScroll: 1
  };
  return (
  <ContentWrapper>
    <CarouselContainer>
      <Slider {...settings}>
        {images.map((image, i) => (
          <ImageContainer key={i}>
            <img src={image} alt='Imagem do ImÃ³vel'/>
          </ImageContainer>
        ))}
      </Slider>
    </CarouselContainer>
    <TextHolder>
      <TextButton
        wrapperStyles={titleWrapperStyles}
        activePortal={activePortal}
        label={title}/>
      {children}
    </TextHolder>
  </ContentWrapper>);
}