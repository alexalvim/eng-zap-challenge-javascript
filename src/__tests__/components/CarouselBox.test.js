import React from 'react';
import { shallow } from 'enzyme';

import { PORTALS } from '../../variables';
import CarouselBox from '../../components/CarouselBox';
import TextButton from '../../components/TextButton';

describe('Testing CarouselBox', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <CarouselBox
        images={['src']}
        title='foo'
        activePortal={PORTALS.ZAP}/>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick function when click in TextButton', () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(
      <CarouselBox
        images={['src']}
        title='foo'
        onClick={mockFunction}
        activePortal={PORTALS.ZAP}/>
    );

    wrapper.find(TextButton).simulate('click');
    expect(mockFunction).toHaveBeenCalled();
  });

  it('should show the passed children', () => {
    const wrapper = shallow(
      <CarouselBox
        images={['src']}
        title='foo'
        activePortal={PORTALS.ZAP}>
        <div id='test'></div>
      </CarouselBox>
    );

    expect(wrapper.find('#test')).toHaveLength(1);
  });
});