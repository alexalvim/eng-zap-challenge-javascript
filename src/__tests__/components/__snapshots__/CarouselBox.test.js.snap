// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CarouselBox should render correctly 1`] = `ShallowWrapper {}`;
